src/scripts/utils.py

```diff
@@ -0,0 +1,280 @@
+        if not (ontology or general_prefix_map or owned_prefix_map):
+            raise ValueError('Term.fetch_missing_props(): Must supply at least one param.')
+
+        # Set: uri, curie
+        prefix_map = general_prefix_map if general_prefix_map else owned_prefix_map
+        if prefix_map and self.uri and not self.curie:
+            converter = self._converter if self._converter else curies.Converter.from_prefix_map(prefix_map)
+            curie = converter.compress(self.uri)
+            self.curie = curie if curie != self.uri else None
+        elif prefix_map and self.curie and not self.uri:
+            converter = curies.Converter.from_prefix_map(prefix_map)
+            uri = converter.expand(self.curie)
+            self.uri = uri if uri != self.curie else None
+
+        # Set: label, definition
+        # todo: When OAK supports: I feel like this would be faster to do outside the Term class itself
+        #  ...e.g. build a dict of CURIE-> Term, then use use ontology.labels() on all curies and set label
+        #  ...Right now, OAK ProntoImplementation labels() is just proxy for label(), though SqlImplementation has, and
+        #  ...Both of them have definition() but not definitions().
+        if ontology:
+            self.label = ontology.label(self.curie)
+            self.definition = ontology.definition(self.curie)
+
+        # Set: direct_parent_uris
+        if ontology and owned_prefix_map:
+            self.direct_parent_uris = self._get_direct_owned_parents(
+                owned_prefix_map=owned_prefix_map if owned_prefix_map else self.owned_prefix_map)
+
+    def __repr__(self):
+        return f'<Term({self.curie if self.curie else self.uri})>'
+
+
+def _load_ontology(ontology_path: str, use_cache=False) -> ProntoImplementation:
+    """Load ontology"""
+    cache_path_ontology = os.path.join(
+        CACHE_DIR, f'{os.path.basename(ontology_path).replace(".owl", "")}.pickle')
+
+    if use_cache and os.path.exists(cache_path_ontology):
+        # todo: Does loading from cache even save time? seemed to take ~16 sec
+        ontology = pickle.load(open(cache_path_ontology, 'rb'))
+    else:
+        # todo's: Ontology implementation selection
+        #  i. If trouble, can try `SparqlImplementation`, but ~6 min to load and queries slow (cuz rdflib)
+        #  ii. use Sql > Pronto when .entities() fixed: https://github.com/INCATools/ontology-access-kit/issues/235
+        # ontology = SqlImplementation(OntologyResource(slug=ontology_path, local=True))
+        ontology = ProntoImplementation(OntologyResource(slug=ontology_path, local=True))  # ~17 sec
+        if use_cache:
+            with open(cache_path_ontology, 'wb') as f:
+                pickle.dump(ontology, f, protocol=pickle.HIGHEST_PROTOCOL)
+
+    return ontology
+
+
+# TODO: implement this func
+# todo: IDs should be int or str? prolly str
+def _get_next_available_mondo_id(min_id: Union[str, int], mondo_ids: Set[int]) -> (int, Set[int]):
```

Comment:
@matentzn Just want to check that I'm implementing this function as you intended:
```py
    next_id = min_id
    while True:
        next_id = next_id + 1
        if next_id not in mondo_ids:
            break
    mondo_ids.add(next_id)
    return next_id, mondo_ids
```

I didn't get too clear on `min_id`. Do I even need to check if it's in `mondo_ids`?

I noticed a bug in this, but I fixed it. I have to re-runt he outputs though. Right now in the new mondo IDs in the outputs have gaps. I was incrementing the ID by 2 instead of 1.

My main concern is this: If I run this on all the ontologies, then there will be overlap in `mondo_id` between all of the output TSV files. They're being assigned the same IDs. If this is an issue, then when I'm running the slurp script, I can check for any `slurp/*.tsv` files and read those IDs in as well and make sure I don't assign IDs that have already been assigned there.

Reply:
Completed new `min_id`, `max_id` (w/ raised error) and reading from `slurp/` changes we discussed.